ibis/pandas/core.py

```diff
@@ -324,6 +381,18 @@ def main_execute(expr, params=None, scope=None, aggcontext=None, **kwargs):
     if scope is None:
         scope = {}
 
+    if timecontext is None:
+        timecontext = {}
+    else:
+        try:
+            timecontext = tuple(map(pd.to_datetime, timecontext))
```

Comment:
I might prohibit ints (and only allow ```Union[pd.Timestamp, str]```)

Reply:
I add a typing in `typing.py` that define timecontext as 
```SupportsTimestamp = Union[pd.Timestamp, str]```
```TimeContext = Optional[Tuple[SupportsTimestamp, SupportsTimestamp]]```